src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -177,33 +194,85 @@ public static ListBoxModel listCheckoutCredentials(SCMSourceOwner context, Strin
 
     public static @Nonnull GitHub connect(@CheckForNull String apiUri, @CheckForNull StandardCredentials credentials) throws IOException {
         String apiUrl = Util.fixEmptyAndTrim(apiUri);
-        String host;
-        try {
-            apiUrl = apiUrl != null ? apiUrl : GitHubServerConfig.GITHUB_URL;
-            host = new URL(apiUrl).getHost();
-        } catch (MalformedURLException e) {
-            throw new IOException("Invalid GitHub API URL: " + apiUrl, e);
-        }
-
-        GitHubBuilder gb = new GitHubBuilder();
-        gb.withEndpoint(apiUrl);
-        gb.withRateLimitHandler(CUSTOMIZED);
-
-        OkHttpClient client = new OkHttpClient().setProxy(getProxy(host));
-
-        gb.withConnector(new OkHttpConnector(new OkUrlFactory(client)));
-
+        apiUrl = apiUrl != null ? apiUrl : GitHubServerConfig.GITHUB_URL;
+        String username;
+        String password;
+        String hash;
         if (credentials == null) {
-            // nothing further to configure
+            username = null;
+            password = null;
+            hash = "anonymous";
         } else if (credentials instanceof StandardUsernamePasswordCredentials) {
             StandardUsernamePasswordCredentials c = (StandardUsernamePasswordCredentials) credentials;
-            gb.withPassword(c.getUsername(), c.getPassword().getPlainText());
+            username = c.getUsername();
+            password = c.getPassword().getPlainText();
+            hash = Util.getDigestOf(password + SALT); // want to ensure pooling by credential
```

Comment:
Yes, I just want to have a way to tell if the password is "substantially" the same for two different routes to the GitHub instance. I do not want to store the actual password, and we only use this for comparison. Anyone who can get a heap dump or access to the JVM sufficient to inspect this would be able to get access to the `Secret` anyway and the salt will be different every time you start up Jenkins so shouldn't be leaking anything much anyway